app/controllers/operations_controller.rb

```diff
@@ -1,2 +1,78 @@
 class OperationsController < ApplicationController
+  before_action :set_operation, only: %i[show edit update destroy]
+  before_action :set_maintenance, only: %i[index create edit update]
+  before_action :set_vehicle, only: %i[index create edit update]
+
+  def index
+    populate_index
+
+    @operation = Operation.new
+    @btn_txt = "Ajouter"
+  end
+
+  def create
+    @operation = Operation.new(operation_params)
+    @operation.maintenance_id = params[:maintenance_id]
+    @operation.maintenance_plan_id = extract_mp_id
+
+    if @operation.save
+      redirect_to vehicle_maintenance_operations_path(params[:vehicle_id], params[:maintenance_id])
+    else
+      @btn_txt = "Ajouter"
+      populate_index
+      render "operations/index"
+    end
+  end
+
+  def edit
+    @btn_txt = 'Editer'
+    populate_index
+    render "operations/index"
+  end
+
+  def update
+    @operation.maintenance_plan_id = extract_mp_id
+    if @operation.update(operation_params)
+      redirect_to vehicle_maintenance_operations_path(params[:vehicle_id], params[:maintenance_id])
```

Comment:
Also here, is there a particular reason of using params for this CRUD?
```suggestion
      redirect_to vehicle_maintenance_operations_path(@vehicle, @maintenance)
```